src/circuit/gadget/sinsemilla/note_commit.rs

```diff
@@ -1058,353 +1107,276 @@ impl NoteCommitConfig {
-                        self.advices[6],
-                        offset,
+                        col_r,
+                        0,
                         || gate_cells.h_1.ok_or(Error::SynthesisError),
                     )?;
+                    CellValue::new(cell, gate_cells.h_1)
+                };
 
-                    // advices[7]
-                    region.assign_advice(
-                        || "d_0",
-                        self.advices[7],
-                        offset,
-                        || gate_cells.d_0.ok_or(Error::SynthesisError),
-                    )?;
+                Ok(h_1)
+            },
+        )?;
 
-                    // advices[8]
-                    copy(
-                        &mut region,
-                        || "b_0",
-                        self.advices[8],
-                        offset,
-                        &gate_cells.b_0,
-                    )?;
+        // |  A_6   | A_7 |   A_8   |     A_9     | q_notecommit_g_d |
+        // -----------------------------------------------------------
+        // | x(g_d) | b_0 | a       | z13_a       |        1         |
+        // |        | b_1 | a_prime | z13_a_prime |        0         |
+        layouter.assign_region(
+            || "NoteCommit input g_d",
+            |mut region| {
+                copy(&mut region, || "gd_x", col_l, 0, &gate_cells.gd_x)?;
 
-                    // advices[9]
-                    region.assign_advice(
-                        || "b_1",
-                        self.advices[9],
-                        offset,
-                        || gate_cells.b_1.ok_or(Error::SynthesisError),
-                    )?;
-                }
+                copy(&mut region, || "b_0", col_m, 0, &gate_cells.b_0)?;
+                copy(&mut region, || "b_1", col_m, 1, &b_1)?;
 
-                // Offset 3
-                {
-                    let offset = 3;
+                copy(&mut region, || "a", col_r, 0, &gate_cells.a)?;
+                copy(&mut region, || "a_prime", col_r, 1, &gate_cells.a_prime)?;
 
-                    // advices[0]
-                    copy(
-                        &mut region,
-                        || "rho",
-                        self.advices[0],
-                        offset,
-                        &gate_cells.rho,
-                    )?;
+                copy(&mut region, || "z13_a", col_z, 0, &gate_cells.z13_a)?;
+                copy(
+                    &mut region,
+                    || "z13_a_prime",
+                    col_z,
+                    1,
+                    &gate_cells.z13_a_prime,
+                )?;
 
-                    // advices[1]
-                    copy(
-                        &mut region,
-                        || "z13_f",
-                        self.advices[1],
-                        offset,
-                        &gate_cells.z13_f,
-                    )?;
+                self.q_notecommit_g_d.enable(&mut region, 0)
```

Comment:
The call to `enable` is before the `copy` calls for other regions; make it consistent.